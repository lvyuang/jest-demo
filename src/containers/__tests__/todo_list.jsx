import React from 'react'
import { mount } from 'enzyme'
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux'

import TodoList from '../todo_list'

describe('业务组件的测试', () => {
  const delay = () => new Promise(resolve => {
    setTimeout(resolve, 0)
  })

  test('render', async () => {
    const fetchSpy = jest.spyOn(window, 'fetch').mockImplementation(() => {
      return new Promise(resolve => {
        resolve({
          json: () => {
            return [
              {id: '1', text: '刷牙'},
              {id: '2', text: '洗脸'},
              {id: '3', text: '吃饭'}
            ]
          }
        })
      })
    })

    const reducer = (state = {
      list: []
    }, action) => {
      switch (action.type) {
        case 'get-list':
          return {
            ...state,
            list: [
              ...state.list, ...action.list
            ]
          }
        default:
          return state
      }
    }
    const store = createStore(reducer, applyMiddleware(thunk))

    const com = mount(
      <Provider store={store}>
        <TodoList />
      </Provider>
    )

    await delay()

    com.update()
    expect(com).toMatchSnapshot()

    fetchSpy.mockRestore()
  })
})