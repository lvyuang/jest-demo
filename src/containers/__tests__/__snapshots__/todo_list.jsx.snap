// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`业务组件的测试 render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(TodoList)>
    <TodoList
      getList={[Function]}
      list={
        Array [
          Object {
            "id": "1",
            "text": "刷牙",
          },
          Object {
            "id": "2",
            "text": "洗脸",
          },
          Object {
            "id": "3",
            "text": "吃饭",
          },
        ]
      }
    >
      <ul>
        <li
          key="1"
        >
          刷牙
        </li>
        <li
          key="2"
        >
          洗脸
        </li>
        <li
          key="3"
        >
          吃饭
        </li>
      </ul>
    </TodoList>
  </Connect(TodoList)>
</Provider>
`;
